pkg/plugins/optional/helm/v2alpha/scaffolds/internal/kustomize/chart_converter.go

```diff
@@ -97,65 +98,176 @@ func (c *ChartConverter) ExtractDeploymentConfig() map[string]interface{} {
 	}
 
 	config := make(map[string]interface{})
+	specMap := extractDeploymentSpec(c.resources.Deployment)
+	if specMap == nil {
+		return config
+	}
 
-	// Extract from deployment spec
-	spec, found, err := unstructured.NestedFieldNoCopy(c.resources.Deployment.Object, "spec", "template", "spec")
-	if !found || err != nil {
+	extractPodSecurityContext(specMap, config)
+
+	container := firstManagerContainer(specMap)
+	if container == nil {
 		return config
 	}
 
+	extractContainerEnv(container, config)
+	extractContainerImage(container, config)
+	extractContainerArgs(container, config)
+	extractContainerResources(container, config)
+	extractContainerSecurityContext(container, config)
```

Comment:
Now we ensure that we extract all values from the manager in the kustomize input.